core-data/reducer.js

```diff
@@ -0,0 +1,13 @@
+const reducer = ( state, action ) => {
+	switch ( action.type ) {
+		case 'FETCH_CATEGORIES_SUCCESS':
```

Comment:
Not sure what your feelings are, but I don't like reducers tracking their data in response to network activity (a successful fetch), and would be inclined toward standalone action types e.g. "receive categories" or similar.

Reply:
I don't have a strong opinion

> I don't like reducers tracking their data in response to network activity (a successful fetch),

The reason for this is probably to stay agnostic to the network but I see the role of this module is to be this adapter between the network and the data needed by the other modules, so might not be too critical.

Reply:
I think this is a common pattern to use `SOMETHING_RECEIVE` when passing the data. See how many occurrences of the word receive are included in Calypso: https://github.com/Automattic/wp-calypso/blob/master/client/state/action-types.js.